01장- 04장/최형우/problem.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * 애플 상품 재고 확인 프로그램 일부
+ *
+ * [문제점]
+ *  - 함수명, 변수명의 모호함과 run함수가 하는 일이 많음.
+ *  - 한 함수에서 여러가지 일을 함
+ *
+ * [개선 방법]
+ *  - 의미 있는 함수명으로 교체
+ *  - run 함수 분리
+ */
+
+import api from './src/api/index.js';
+import { helper } from './src/utils/index.js';
+import db from './src/db/index.js';
+
+// 우편번호
+const LOCATION = '05803';
+
+/**
+ * 상품 구매 가능 여부 가져오기
+ * @param {string} sku
+ * @param {string} location
+ * @returns {bool}
+ *
+ * 문제점
+ *  - code 파라미터 명의 모호함
+ *  - 함수명은 '재고 정보 패치 받기'지만 실제로 '구매 가능'여부를 리턴함
+ */
+async function fetchStockInfo(sku, location) {
```

Comment:
sku가 약어 일 것 같다는 생각이 드는데, location과 상품 코드 인지,  stock 어쩌고의 약자인지 코드만 보고는 한눈에 안들어오는게 아쉽네요! 조금 인자명이 길더라도 인수만 보고 확실히 알 수 있으면 좋을 것 같아요 
```suggestion
 * 상품 구매 가능 여부 가져오기
 * @param {string} sku
 * @param {string} location
 * @returns {bool}
 *
 * 문제점
 *  - code 파라미터 명의 모호함
 *  - 함수명은 '재고 정보 패치 받기'지만 실제로 '구매 가능'여부를 리턴함
 */
async function fetchStockInfo(sku, location) {
```

Reply:
애플 API에서 실제로 sku와 location이라고 사용하고 있기에 해당 네이밍을 따라갔습니다. 
그리고 만약 팀 내에서 이런 코드를 짠다면 이미 팀원들이 이 단어들을 알고 있을거라 생각해서 따로 주석이나 변수명을 통해 설명을 적어두진 않았습니다. 하지만 개인프로젝트니 설명이라도 써두는게 맞는 것 같네요 ㅎㅎ